services/worker-manager/src/providers/aws.js

```diff
@@ -108,14 +105,16 @@ class AwsProvider extends Provider {
         });
       }
 
+      const instanceCount = Math.ceil(Math.min(toSpawnCounter, toSpawnPerConfig) / config.capacityPerInstance);
+      let spawned;
       try {
         spawned = await this.ec2s[config.region].runInstances({
           ...config.launchConfig,
 
           UserData: userData.toString('base64'), // The string needs to be base64-encoded. See the docs above
 
-          MaxCount: Math.min(toSpawnCounter, toSpawnPerConfig),
-          MinCount: Math.min(toSpawnCounter, toSpawnPerConfig),
+          MaxCount: instanceCount,
```

Comment:
I'm probably not being clear. I meant:

```
MaxCount: Math.min(Math.ceil(toSpawnCounter  / config.capacityPerInstance), Math.ceil(toSpawnPerConfig / config.capacityPerInstance) )
```

in `runInstances`, and the same for `minCount`. Why don't I just test what I am suggesting tho? Maybe it doesn't work

Reply:
The `instanceCount` calculations are equivalent to this afaict.